src/pages/userManagement/components/EditCollaborationDialog.jsx

```diff
@@ -0,0 +1,224 @@
+
+import Alert from '../../../components/Alert';
+import Button from '../../../components/Button';
+import InputRow from '../../../components/fields/edit/InputRow';
+import SelectionEditor from '../../../components/fields/edit/SelectionEditor';
+import StandardDialog from '../../../components/StandardDialog';
+import Text from '../../../components/Text';
+import usePatchCollaboration from '../../../models/collaboration/usePatchCollaboration';
+import {
+  getSummaryState,
+  isEditApproved,
+  isViewApproved,
+  isViewRevoked,
+  states as collaborationStates,
+  summaryStates,
+} from '../../../models/collaboration/model';
+
+const coreStateChoices = [
+  {
+    value: 'view',
+    labelId: 'COLLABORATION_STATE_VIEW',
+  },
+  {
+    value: 'edit',
+    labelId: 'COLLABORATION_STATE_EDIT',
+  },
+];
+
+const fullStateChoices = [
+  ...coreStateChoices,
+  {
+    value: 'revoked',
+    labelId: 'COLLABORATION_STATE_REVOKED',
+  },
+];
+
+function FormattedUserName({ name }) {
+  return name || <FormattedMessage id="UNNAMED_USER" />;
+}
+
+function getPatchCollaborationOperations(
+  newSummaryState,
+  collaboration,
+) {
+  switch (newSummaryState) {
+    case summaryStates.revoked:
+      return [
+        {
+          op: 'replace',
+          path: '/managed_view_permission',
+          value: {
+            permission: collaborationStates.revoked,
+          },
+        },
+      ];
+    case summaryStates.edit:
+      return [
+        {
+          op: 'replace',
+          path: '/managed_edit_permission',
+          value: {
+            permission: collaborationStates.approved,
+          },
+        },
+      ];
+    case summaryStates.view: {
+      const operations = [
+        {
+          op: 'replace',
+          path: '/managed_view_permission',
+          value: {
+            permission: collaborationStates.approved,
+          },
+        },
+      ];
+
+      if (isEditApproved(collaboration)) {
+        operations.push({
+          op: 'replace',
+          path: '/managed_edit_permission',
+          value: {
+            permission: collaborationStates.revoked,
+          },
+        });
+      }
+
+      return operations;
+    }
+    default:
+      return [];
+  }
+}
+
+export default function EditCollaborationDialog({
+  collaboration,
+  open,
+  onClose,
+}) {
+  const [collaborationState, setCollaborationState] = useState(
+    getSummaryState(collaboration),
+  );
+  const members = Object.values(collaboration?.members || {});
+
+  const {
+    mutate: patchCollaboration,
+    isLoading,
+    error,
+    clearError,
+  } = usePatchCollaboration();
+
+  useEffect(() => {
+    setCollaborationState(getSummaryState(collaboration));
+  }, [collaboration]);
+
+  const stateChoices =
+    isViewApproved(collaboration) || isViewRevoked(collaboration)
+      ? fullStateChoices
+      : coreStateChoices;
+
+  function handleClose() {
+    // cleanup
+    setCollaborationState('');
+    clearError();
+
+    // close
```

Comment:
these comments arent really necessary

Reply:
Removed in https://github.com/WildMeOrg/codex-frontend/pull/451/commits/0e0918d3e8d4c5901242ff18c8f4edecf9b494b3.